src/App.js

```diff
@@ -6,31 +6,45 @@ function App() {
   const [todoList, setTodoList] = useState([]); //updated to empty array
   const [isLoading, setIsLoading] = useState(true);
 
+  const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/Default`;
+  async function fetchData() {
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
+    };
+    try {
+      const response = await fetch(url, options);
+      if (!response.ok) {
+        const message = `Error occured: ${response.status}`;
+        throw new Error(message);
+      }
+      const data = await response.json();
+      const todosMain = data.records.map((todo) => {
+        const newTodo = {
+          title: todo.fields.title,
+          id: todo.id,
+        };
+        return newTodo;
+      });
+      setTodoList(todosMain);
+    } catch (error) {
+      console.log(error.message);
+    }
+  }
   useEffect(() => {
-    new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem("SavedTodoList")) ?? [],
-          },
-        });
-      }, 2000);
-    }).then((result) => {
-      setTodoList(result.data.todoList);
-      setIsLoading(false);
-    });
+    fetchData();
+    setIsLoading(false);
```

Comment:
Requires change:
This is a racing condition where fetching data and setting the `isLoading` state to false is not guaranteed to be set before the data has been fetched. Instead, this `setIsLoading` should be called after the `setTodoList` on line 31.

Reply:
This useEffect is missing `fetchData` as a dependency, however when defining functions within a component (like `App`) body, a function as a dependency must be wrapped in a [useCallback](https://react.dev/reference/react/useCallback) to avoid re-renders. Therefore, I'd recommend that you...

either define the `fetchData` within the useEffect body and invoke it:
```js
useEffect(() => {
  const fetchData = async () => { /* insert fetch data code here */ }
  fetchData();
}, []);
```
OR, wrap `fetchData` with a `useCallback` function and add it a dependency:
```js
import {useCallback, useState, useEffect} from 'react';

function App() { 
  ...

  const fetchData = useCallback(async () => { /* insert fetch data code here */ }, []);

  useEffect(() => {
    fetchData();
  }, [fetchData]); // add fetchData as a dependency here

  return (...)
}
```